File: API/author-service/src/main/java/com/friendbook/service/AuthorService.java
package com.friendbook.service;

import java.util.List;

import org.springframework.data.domain.Page;

import com.friendbook.DTO.AuthorForBook;
import com.friendbook.DTO.AuthorForSearch;
import com.friendbook.DTO.AuthorWithBooks;
import com.friendbook.model.Author;
import com.friendbook.utils.Sort;

public interface AuthorService {
    Author getAuthor(Long id);

    AuthorWithBooks getAuthorWithBooks(Long authorId);

    AuthorForBook getAuthorForBook(Long authorId);

    List<AuthorForSearch> getAuthorsByAuthorName(String name);

    List<AuthorForSearch> getAllAuthorsForSearch();

    Page<AuthorForSearch> search(int numberPage, int sizePage, Sort sort, String word, int startRating, int finishRating);
}

File: client/app/src/main/java/com/src/book/data/remote/service/AuthorService.kt
package com.src.book.data.remote.service

import com.src.book.data.remote.model.author.author.AuthorResponse
import com.src.book.data.remote.model.author.authorList.AuthorListResponse
import com.src.book.data.remote.model.author.authorSearchResult.AuthorSearchResultResponse
import com.src.book.utils.AUTHOR_SERVICE_BASE_URL
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface AuthorService {
    @GET("${AUTHOR_SERVICE_BASE_URL}/author/{id}")
    suspend fun getAuthorById(@Path("id") id: Long): Response<AuthorResponse>

    @GET("$AUTHOR_SERVICE_BASE_URL/author/search")
    suspend fun searchAuthors(
        @Query("numberPage") numberPage: Int,
        @Query("sizePage") sizePage: Int,
        @Query("word") word: String?,
        @Query("sort") sort: String?,
        @Query("startRating") startRating: Int?,
        @Query("finishRating") finishRating: Int?
    ): Response<AuthorSearchResultResponse>
}